MariaDB egg broken? Cannot get it working.

I originally tested with this rootless podman. I was trying to create and deploy a mariadb server in Pterodactyl. I was faced with permission errors from inside the container, which seemed weird to me. I originally thought this was a podman problem but then I tried running the image with docker and I got the same issue. So I am wondering if I'm doing something really wrong or if the image is broken for both docker and podman. The issue seems to be relating to the user inside the container not being able to overwrite the `/tmp/passwd` and `/tmp/group` files. The user inside the container is called `container` and has uid and gid `1000`. 

Btw the egg/image in question is the mariadb egg, and specifically this image [ghcr.io/ptero-eggs/yolks:mariadb_11.6](https://github.com/ptero-eggs/yolks/pkgs/container/yolks/370608927?tag=mariadb_11.6). I also tested with the 11.4 image and got the same error.

### Extra Context:
- For testing with Podman, I ran the image in Podman 5.4.1
- For testing with Docker, I ran the image on a different machine on Docker version 20.10.24
- Both machines were rented Ryzen 9 7950X3D machines from Hetzner with Debian 12 as the host OS.

## Error Log in Pterodactyl.
Note: Wings is deploying rootless podman containers instead of deploying docker containers.
```
[Pterodactyl Daemon]: Finished pulling Docker container image
/entrypoint.sh: line 27: /tmp/passwd: Permission denied
/entrypoint.sh: line 28: /tmp/group: Permission denied
Starting MariaDB server...
2025-03-12 16:11:41 0 [Note] Starting MariaDB 11.6.2-MariaDB-ubu2404 source revision d8dad8c3b54cd09fefce7bc3b9749f427eed9709 server_uid x0tvKozUeGNVi2uoZT3LtiQNRtI= as process 13
2025-03-12 16:11:41 0 [Note] InnoDB: Compressed tables use zlib 1.3
2025-03-12 16:11:41 0 [Note] InnoDB: Number of transaction pools: 1
2025-03-12 16:11:41 0 [Note] InnoDB: Using AVX512 instructions
2025-03-12 16:11:41 0 [Warning] mariadbd: io_uring_queue_init() failed with errno 0
2025-03-12 16:11:41 0 [Warning] InnoDB: liburing disabled: falling back to innodb_use_native_aio=OFF
2025-03-12 16:11:41 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2025-03-12 16:11:41 0 [Note] InnoDB: Completed initialization of buffer pool
2025-03-12 16:11:41 0 [Note] InnoDB: File system buffers for log disabled (block size=4096 bytes)
2025-03-12 16:11:41 0 [Note] InnoDB: Starting crash recovery from checkpoint LSN=46316
2025-03-12 16:11:41 0 [Note] InnoDB: End of log at LSN=48547
2025-03-12 16:11:41 0 [Note] InnoDB: To recover: 22 pages
2025-03-12 16:11:41 0 [Note] InnoDB: Opened 3 undo tablespaces
2025-03-12 16:11:41 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
2025-03-12 16:11:41 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
2025-03-12 16:11:41 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2025-03-12 16:11:41 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2025-03-12 16:11:41 0 [Note] InnoDB: log sequence number 48547; transaction id 15
2025-03-12 16:11:41 0 [Note] InnoDB: Loading buffer pool(s) from /home/container/mysql/ib_buffer_pool
2025-03-12 16:11:41 0 [Note] Plugin 'FEEDBACK' is disabled.
2025-03-12 16:11:41 0 [Note] Plugin 'wsrep-provider' is disabled.
2025-03-12 16:11:41 0 [Note] Recovering after a crash using tc.log
2025-03-12 16:11:41 0 [Note] Starting table crash recovery...
2025-03-12 16:11:41 0 [Note] Crash table recovery finished.
2025-03-12 16:11:41 0 [Note] InnoDB: Buffer pool(s) load completed at 250312 16:11:41
2025-03-12 16:11:42 0 [Note] Server socket created on IP: '0.0.0.0'.
2025-03-12 16:11:42 0 [Warning] 'user' entry '@4f119c14-a8f8-4705-8ea8-149cc5cf0d75' ignored in --skip-name-resolve mode.
2025-03-12 16:11:42 0 [Warning] 'proxies_priv' entry '@% container@4f119c14-a8f8-4705-8ea8-149cc5cf0d75' ignored in --skip-name-resolve mode.
2025-03-12 16:11:42 0 [Note] mariadbd: Event Scheduler: Loaded 0 events
2025-03-12 16:11:42 0 [Note] mariadbd: ready for connections.
Version: '11.6.2-MariaDB-ubu2404'  socket: '/home/container/run/mysqld/mysqld.sock'  port: 25520  mariadb.org binary distribution
container@pterodactyl~ Server marked as running...
Upgrading MariaDB if needed...
2025-03-12 16:11:46 3 [Warning] Access denied for user 'container'@'localhost'
Reading datadir from the MariaDB server failed. Got the following error when executing the 'mysql' command line client
ERROR 1698 (28000): Access denied for user 'container'@'localhost'
FATAL ERROR: Upgrade failed
:/home/container$ ${MARIADB_EXECUTABLE}
250312 16:11:46 [ERROR] mysqld got signal 11 ;
Sorry, we probably made a mistake, and this is a bug.
Your assistance in bug reporting will enable us to fix this for the next release.
To report this bug, see https://mariadb.com/kb/en/reporting-bugs
We will try our best to scrape up some info that will hopefully help
diagnose the problem, but since we have already crashed, 
something is definitely wrong and this may fail.
Server version: 11.6.2-MariaDB-ubu2404 source revision: d8dad8c3b54cd09fefce7bc3b9749f427eed9709
key_buffer_size=134217728
read_buffer_size=131072
max_used_connections=1
max_threads=153
thread_count=1
It is possible that mysqld could use up to 
key_buffer_size + (read_buffer_size + sort_buffer_size)*max_threads = 468085 K  bytes of memory
Hope that's ok; if not, decrease some variables in the equation.
Thread pointer: 0x7fbab8000c68
Attempting backtrace. You can use the following information to find out
where mysqld died. If you see no messages after this, something went
terribly wrong...
stack_bottom = 0x7fbaf80cf000 thread_stack 0x49000
Printing to addr2line failed
mariadbd(my_print_stacktrace+0x32)[0x5556bd966472]
mariadbd(handle_fatal_signal+0x2c3)[0x5556bd40a9c3]
/lib/x86_64-linux-gnu/libc.so.6(+0x45320)[0x7fbb0f8e6320]
/lib/x86_64-linux-gnu/libc.so.6(+0x19d0a5)[0x7fbb0fa3e0a5]
mariadbd(+0x10fda0a)[0x5556bd9dea0a]
mariadbd(+0x7de689)[0x5556bd0bf689]
mariadbd(_Z16acl_authenticateP3THDj+0x492)[0x5556bd0e6ac2]
mariadbd(+0x9d2cb5)[0x5556bd2b3cb5]
mariadbd(_Z22thd_prepare_connectionP3THD+0x4e)[0x5556bd2b450e]
mariadbd(_Z24do_handle_one_connectionP7CONNECTb+0x36a)[0x5556bd2b4fda]
mariadbd(handle_one_connection+0x6d)[0x5556bd2b535d]
mariadbd(+0xd736a4)[0x5556bd6546a4]
/lib/x86_64-linux-gnu/libc.so.6(+0x9ca94)[0x7fbb0f93da94]
/lib/x86_64-linux-gnu/libc.so.6(+0x129c3c)[0x7fbb0f9cac3c]
Trying to get some variables.
Some pointers may be invalid and cause the dump to abort.
Query (0x0): (null)
Connection ID (thread ID): 4
Status: NOT_KILLED
Optimizer switch: index_merge=on,index_merge_union=on,index_merge_sort_union=on,index_merge_intersection=on,index_merge_sort_intersection=off,index_condition_pushdown=on,derived_merge=on,derived_with_keys=on,firstmatch=on,loosescan=on,materialization=on,in_to_exists=on,semijoin=on,partial_match_rowid_merge=on,partial_match_table_scan=on,subquery_cache=on,mrr=off,mrr_cost_based=off,mrr_sort_keys=off,outer_join_with_cache=on,semijoin_with_cache=on,join_cache_incremental=on,join_cache_hashed=on,join_cache_bka=on,optimize_join_buffer_size=on,table_elimination=on,extended_keys=on,exists_to_in=on,orderby_uses_equalities=on,condition_pushdown_for_derived=on,split_materialized=on,condition_pushdown_for_subquery=on,rowid_filter=on,condition_pushdown_from_having=on,not_null_range_scan=off,hash_join_cardinality=on,cset_narrowing=off,sargable_casefold=on
The manual page at https://mariadb.com/kb/en/how-to-produce-a-full-stack-trace-for-mariadbd/ contains
information that should help you find out what is causing the crash.
We think the query pointer is invalid, but we will try to print it anyway. 
Query: 
Writing a core file...
Working directory at /home/container/mysql
Resource Limits:
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            8388608              unlimited            bytes     
Max core file size        0                    unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             511451               511451               processes 
Max open files            1048576              1048576              files     
Max locked memory         16765906944          16765906944          bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       511451               511451               signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us        
Core pattern: core
Kernel version: Linux version 6.1.0-31-amd64 (debian-kernel@lists.debian.org) (gcc-12 (Debian 12.2.0-14) 12.2.0, GNU ld (GNU Binutils for Debian) 2.40) #1 SMP PREEMPT_DYNAMIC Debian 6.1.128-1 (2025-02-07)
ERROR 2026 (HY000): TLS/SSL error: unexpected eof while reading
/entrypoint.sh: line 64:    13 Segmentation fault      $MARIADBD_EXECUTABLE --datadir=$HOME/mysql
```


## Inspecting the `entrypoint.sh` file
Lines 27 and 28 of this file are:
`envsubst < /passwd.template > ${NSS_WRAPPER_PASSWD}`
`envsubst < /group.template > ${NSS_WRAPPER_GROUP}`
```bash
#!/bin/bash
cd /home/container

# Make internal Docker IP address available to processes.
INTERNAL_IP=$(ip route get 1 | awk '{print $(NF-2);exit}')
export INTERNAL_IP

# Determine correct executables to use
if which mariadbd-safe > /dev/null ; then
	MARIADBD_EXECUTABLE="mariadbd"
	MARIADB_INSTALLDB_EXECUTABLE="mariadb-install-db"
	MARIAAB_UPGRADE_EXECUTABLE="mariadb-upgrade"
	MARIADB_EXECUTABLE="mariadb"

else
	MARIADBD_EXECUTABLE="mysqld"
	MARIADB_INSTALLDB_EXECUTABLE="mysql_install_db"
	MARIAAB_UPGRADE_EXECUTABLE="mysql_upgrade"
	MARIADB_EXECUTABLE="mysql"
fi



# Setup NSS Wrapper - Some valiables already set in the Dockerfile
export USER_ID=$(id -u)
export GROUP_ID=$(id -g)
envsubst < /passwd.template > ${NSS_WRAPPER_PASSWD}
envsubst < /group.template > ${NSS_WRAPPER_GROUP}
if [ -f  /usr/lib/libnss_wrapper.so ]; then
	export LD_PRELOAD=/usr/lib/libnss_wrapper.so
else
	export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libnss_wrapper.so
fi


# Ensure required folders exist
mkdir -p $HOME/run/mysqld
mkdir -p $HOME/log/mysql
mkdir -p $HOME/mysql


# Only run install db if the database is not setup
if [ ! -f mysql/mysql/user.frm ] ; then
	echo "Database not setup - running initial setup"
	$MARIADB_INSTALLDB_EXECUTABLE --datadir=$HOME/mysql
fi


echo "Starting MariaDB server..."
$MARIADBD_EXECUTABLE --datadir=$HOME/mysql &
PID=$!

sleep 5

echo "Upgrading MariaDB if needed..."
$MARIAAB_UPGRADE_EXECUTABLE -u container


# Replace Startup Variables
MODIFIED_STARTUP=$(echo ${STARTUP} | sed -e 's/{{/${/g' -e 's/}}/}/g')
echo ":/home/container$ ${MODIFIED_STARTUP}"

# Run the Server
eval ${MODIFIED_STARTUP}

# Wait for the server to stop
wait $PID > /dev/null 2>&1
```


## Inspecting inside the container
```bash
minecraft@Ryzen9-Debian12 ~ $ podman run --entrypoint /bin/sh -it 8a12e0341e85
$ cd /

$ ls
bin                boot  docker-entrypoint-initdb.d  etc             home  lib.usr-is-merged  media  opt              proc  run   sbin.usr-is-merged  sys  usr
bin.usr-is-merged  dev   entrypoint.sh               group.template  lib   lib64              mnt    passwd.template  root  sbin  srv                 tmp  var

$ ls -lah /tmp
total 8.0K
drwxrwxrwt 1 root root 4.0K Mar 10 00:39 .
dr-xr-xr-x 1 root root 4.0K Mar 13 15:00 ..
-rw-rw-r-- 1 root root    0 Mar 10 00:39 group
-rw-rw-r-- 1 root root    0 Mar 10 00:39 passwd

$ cat passwd.template
root:x:0:0:root:/root:/bin/bash
${USER}:x:${USER_ID}:${GROUP_ID}:${USER}:${HOME}:/bin/bash

$ echo ${NSS_WRAPPER_PASSWD}
/tmp/passwd

$ envsubst < /passwd.template > ${NSS_WRAPPER_PASSWD}
/bin/sh: 6: cannot create /tmp/passwd: Permission denied

$ rm -rf /tmp/passwd
rm: cannot remove '/tmp/passwd': Operation not permitted

$ ls -lah /tmp
total 8.0K
drwxrwxrwt 1 root root 4.0K Mar 10 00:39 .
dr-xr-xr-x 1 root root 4.0K Mar 13 15:00 ..
-rw-rw-r-- 1 root root    0 Mar 10 00:39 group
-rw-rw-r-- 1 root root    0 Mar 10 00:39 passwd

$
```
User `container` isn't able to overwrite the files in the `/tmp` directory. However the `container` user is able to create new files in the `/tmp` directory, just not able to overwrite or remove ones owned by root. Which makes sense because the files in that directory have `664` perms and the owner and group of the files is `root:root` with `container` not being in the `root` group.



## Same initial error when running with Docker
My first thinking was that it seems to be broken because I'm using podman but then I tried it on an entirely different machine with Docker instead of Podman and I got pretty much the same issue. I also checked the perms inside the container and the `container` user is still not able to overwrite the `/tmp` directory files.
```
user_dom@GER-UM-02-7950X3D:~$ docker run -it 8a12e0341e85
/entrypoint.sh: line 27: /tmp/passwd: Permission denied
/entrypoint.sh: line 28: /tmp/group: Permission denied
Database not setup - running initial setup
Installing MariaDB/MySQL system tables in '/home/container/mysql' ...
OK

To start mariadbd at boot time you have to copy
support-files/mariadb.service to the right place for your system


Two all-privilege accounts were created.
One is root@localhost, it has no password, but you need to
be system 'root' user to connect. Use, for example, sudo mysql
The second is container@localhost, it has no password either, but
you need to be the system 'container' user to connect.
After connecting you can set the password, if you would need to be
able to connect as any of these users with a password and without sudo

See the MariaDB Knowledgebase at https://mariadb.com/kb

You can start the MariaDB daemon with:
cd '/usr' ; /usr/bin/mariadbd-safe --datadir='/home/container/mysql'

You can test the MariaDB daemon with mariadb-test-run.pl
cd '/usr/share/mariadb/mariadb-test' ; perl mariadb-test-run.pl

Please report any problems at https://mariadb.org/jira

The latest information about MariaDB is available at https://mariadb.org/.

Consider joining MariaDB's strong and vibrant community:
https://mariadb.org/get-involved/

Starting MariaDB server...
2025-03-13 15:35:48 0 [Note] Starting MariaDB 11.6.2-MariaDB-ubu2404 source revision d8dad8c3b54cd09fefce7bc3b9749f427eed9709 server_uid Tnq7keT8GKDxmAtaGW699QQYKjk= as process 70
2025-03-13 15:35:48 0 [Note] InnoDB: Compressed tables use zlib 1.3
2025-03-13 15:35:48 0 [Note] InnoDB: Number of transaction pools: 1
2025-03-13 15:35:48 0 [Note] InnoDB: Using AVX512 instructions
2025-03-13 15:35:48 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2025-03-13 15:35:48 0 [Note] InnoDB: Using liburing
2025-03-13 15:35:48 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2025-03-13 15:35:48 0 [Note] InnoDB: Completed initialization of buffer pool
2025-03-13 15:35:48 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
2025-03-13 15:35:48 0 [Note] InnoDB: End of log at LSN=46300
2025-03-13 15:35:48 0 [Note] InnoDB: Opened 3 undo tablespaces
2025-03-13 15:35:48 0 [Note] InnoDB: 128 rollback segments in 3 undo tablespaces are active.
2025-03-13 15:35:48 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2025-03-13 15:35:48 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2025-03-13 15:35:48 0 [Note] InnoDB: log sequence number 46300; transaction id 14
2025-03-13 15:35:48 0 [Note] InnoDB: Loading buffer pool(s) from /home/container/mysql/ib_buffer_pool
2025-03-13 15:35:48 0 [Note] Plugin 'FEEDBACK' is disabled.
2025-03-13 15:35:48 0 [Note] Plugin 'wsrep-provider' is disabled.
2025-03-13 15:35:48 0 [Note] InnoDB: Buffer pool(s) load completed at 250313 15:35:48
2025-03-13 15:35:49 0 [Note] Server socket created on IP: '0.0.0.0'.
2025-03-13 15:35:49 0 [Note] Server socket created on IP: '::'.
2025-03-13 15:35:49 0 [Warning] 'user' entry '@27f09f403e9f' ignored in --skip-name-resolve mode.
2025-03-13 15:35:49 0 [Warning] 'proxies_priv' entry '@% container@27f09f403e9f' ignored in --skip-name-resolve mode.
2025-03-13 15:35:49 0 [Note] mariadbd: Event Scheduler: Loaded 0 events
2025-03-13 15:35:49 0 [Note] mariadbd: ready for connections.
Version: '11.6.2-MariaDB-ubu2404'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
Upgrading MariaDB if needed...
2025-03-13 15:35:53 3 [Warning] Access denied for user 'container'@'localhost'
Reading datadir from the MariaDB server failed. Got the following error when executing the 'mysql' command line client
ERROR 1698 (28000): Access denied for user 'container'@'localhost'
FATAL ERROR: Upgrade failed
:/home/container$
```

--- a/mariadb/10.11/Dockerfile
+++ b/mariadb/10.11/Dockerfile
@@ -8,30 +8,25 @@
 #
 # ---------------------------------------------------------
 
-	FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.11
+FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.11
 
-	LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
+LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
 
-	ENV         DEBIAN_FRONTEND noninteractive
+ENV         DEBIAN_FRONTEND=noninteractive
 
-	RUN         apt update -y && apt install -y libnss-wrapper gettext-base
-	RUN 		useradd -d /home/container -m container -s /bin/bash
-
-
-
-	## Prepare NSS Wrapper for the entrypoint
-	ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
-	RUN         touch ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chgrp 0 ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chmod g+rw ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP}
-	ADD         nss/passwd.template /passwd.template
-	ADD         nss/group.template /group.template
+RUN         apt update -y && apt install -y libnss-wrapper gettext-base
+RUN 		useradd -d /home/container -m container -s /bin/bash
 
 
-	USER        container
-	ENV         USER=container HOME=/home/container
-	WORKDIR     /home/container
+## Prepare NSS Wrapper for the entrypoint
+ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
+COPY		--chown=root:root --chmod=644 nss/passwd.template nss/group.template /
 
 
-	COPY 		--chmod=755 ../entrypoint.sh /entrypoint.sh
-	ENTRYPOINT ["/entrypoint.sh"]
+USER        container
+ENV         USER=container HOME=/home/container
+WORKDIR     /home/container
+
+
+COPY		--chown=root:root --chmod=755 ../entrypoint.sh /entrypoint.sh
+ENTRYPOINT	["/entrypoint.sh"]

--- a/mariadb/10.5/Dockerfile
+++ b/mariadb/10.5/Dockerfile
@@ -8,30 +8,25 @@
 #
 # ---------------------------------------------------------
 
-	FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.5
+FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.5
 
-	LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
+LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
 
-	ENV         DEBIAN_FRONTEND noninteractive
+ENV         DEBIAN_FRONTEND=noninteractive
 
-	RUN         apt update -y && apt install -y libnss-wrapper gettext-base
-	RUN 		useradd -d /home/container -m container -s /bin/bash
-
-
-
-	## Prepare NSS Wrapper for the entrypoint
-	ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
-	RUN         touch ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chgrp 0 ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chmod g+rw ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP}
-	ADD         nss/passwd.template /passwd.template
-	ADD         nss/group.template /group.template
+RUN         apt update -y && apt install -y libnss-wrapper gettext-base
+RUN 		useradd -d /home/container -m container -s /bin/bash
 
 
-	USER        container
-	ENV         USER=container HOME=/home/container
-	WORKDIR     /home/container
+## Prepare NSS Wrapper for the entrypoint
+ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
+COPY		--chown=root:root --chmod=644 nss/passwd.template nss/group.template /
 
 
-	COPY 		--chmod=755 ../entrypoint.sh /entrypoint.sh
-	ENTRYPOINT ["/entrypoint.sh"]
+USER        container
+ENV         USER=container HOME=/home/container
+WORKDIR     /home/container
+
+
+COPY		--chown=root:root --chmod=755 ../entrypoint.sh /entrypoint.sh
+ENTRYPOINT	["/entrypoint.sh"]

--- a/mariadb/10.6/Dockerfile
+++ b/mariadb/10.6/Dockerfile
@@ -8,30 +8,25 @@
 #
 # ---------------------------------------------------------
 
-	FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.6
+FROM        --platform=$TARGETOS/$TARGETARCH mariadb:10.6
 
-	LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
+LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
 
-	ENV         DEBIAN_FRONTEND noninteractive
+ENV         DEBIAN_FRONTEND=noninteractive
 
-	RUN         apt update -y && apt install -y libnss-wrapper gettext-base
-	RUN 		useradd -d /home/container -m container -s /bin/bash
-
-
-
-	## Prepare NSS Wrapper for the entrypoint
-	ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
-	RUN         touch ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chgrp 0 ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chmod g+rw ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP}
-	ADD         nss/passwd.template /passwd.template
-	ADD         nss/group.template /group.template
+RUN         apt update -y && apt install -y libnss-wrapper gettext-base
+RUN 		useradd -d /home/container -m container -s /bin/bash
 
 
-	USER        container
-	ENV         USER=container HOME=/home/container
-	WORKDIR     /home/container
+## Prepare NSS Wrapper for the entrypoint
+ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
+COPY		--chown=root:root --chmod=644 nss/passwd.template nss/group.template /
 
 
-	COPY 		--chmod=755 ../entrypoint.sh /entrypoint.sh
-	ENTRYPOINT ["/entrypoint.sh"]
+USER        container
+ENV         USER=container HOME=/home/container
+WORKDIR     /home/container
+
+
+COPY		--chown=root:root --chmod=755 ../entrypoint.sh /entrypoint.sh
+ENTRYPOINT	["/entrypoint.sh"]

--- a/mariadb/11.4/Dockerfile
+++ b/mariadb/11.4/Dockerfile
@@ -8,30 +8,25 @@
 #
 # ---------------------------------------------------------
 
-	FROM        --platform=$TARGETOS/$TARGETARCH mariadb:11.4
+FROM        --platform=$TARGETOS/$TARGETARCH mariadb:11.4
 
-	LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
+LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
 
-	ENV         DEBIAN_FRONTEND noninteractive
+ENV         DEBIAN_FRONTEND=noninteractive
 
-	RUN         apt update -y && apt install -y libnss-wrapper gettext-base
-	RUN 		useradd -d /home/container -m container -s /bin/bash
-
-
-
-	## Prepare NSS Wrapper for the entrypoint
-	ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
-	RUN         touch ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chgrp 0 ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chmod g+rw ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP}
-	ADD         nss/passwd.template /passwd.template
-	ADD         nss/group.template /group.template
+RUN         apt update -y && apt install -y libnss-wrapper gettext-base
+RUN 		useradd -d /home/container -m container -s /bin/bash
 
 
-	USER        container
-	ENV         USER=container HOME=/home/container
-	WORKDIR     /home/container
+## Prepare NSS Wrapper for the entrypoint
+ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
+COPY		--chown=root:root --chmod=644 nss/passwd.template nss/group.template /
 
 
-	COPY 		--chmod=755 ../entrypoint.sh /entrypoint.sh
-	ENTRYPOINT ["/entrypoint.sh"]
+USER        container
+ENV         USER=container HOME=/home/container
+WORKDIR     /home/container
+
+
+COPY		--chown=root:root --chmod=755 ../entrypoint.sh /entrypoint.sh
+ENTRYPOINT	["/entrypoint.sh"]

--- a/mariadb/11.6/Dockerfile
+++ b/mariadb/11.6/Dockerfile
@@ -8,30 +8,25 @@
 #
 # ---------------------------------------------------------
 
-	FROM        --platform=$TARGETOS/$TARGETARCH mariadb:11.6
+FROM        --platform=$TARGETOS/$TARGETARCH mariadb:11.6
 
-	LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
+LABEL       author="Daniel Barton" maintainer="danielb@purpleflaghosting.com"
 
-	ENV         DEBIAN_FRONTEND noninteractive
+ENV         DEBIAN_FRONTEND=noninteractive
 
-	RUN         apt update -y && apt install -y libnss-wrapper gettext-base
-	RUN 		useradd -d /home/container -m container -s /bin/bash
-
-
-
-	## Prepare NSS Wrapper for the entrypoint
-	ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
-	RUN         touch ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chgrp 0 ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP} \
-				&& chmod g+rw ${NSS_WRAPPER_PASSWD} ${NSS_WRAPPER_GROUP}
-	ADD         nss/passwd.template /passwd.template
-	ADD         nss/group.template /group.template
+RUN         apt update -y && apt install -y libnss-wrapper gettext-base
+RUN 		useradd -d /home/container -m container -s /bin/bash
 
 
-	USER        container
-	ENV         USER=container HOME=/home/container
-	WORKDIR     /home/container
+## Prepare NSS Wrapper for the entrypoint
+ENV         NSS_WRAPPER_PASSWD=/tmp/passwd NSS_WRAPPER_GROUP=/tmp/group
+COPY		--chown=root:root --chmod=644 nss/passwd.template nss/group.template /
 
 
-	COPY 		--chmod=755 ../entrypoint.sh /entrypoint.sh
-	ENTRYPOINT ["/entrypoint.sh"]
+USER        container
+ENV         USER=container HOME=/home/container
+WORKDIR     /home/container
+
+
+COPY		--chown=root:root --chmod=755 ../entrypoint.sh /entrypoint.sh
+ENTRYPOINT	["/entrypoint.sh"]

